DhsTranslator/src/DhsAdapter.cpp

```diff
@@ -0,0 +1,178 @@
+            switch (iter->getType()) {
+            case DHS_DT_UINT8: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<uint8_t>(), &status);
+                break;
+            }
+            case DHS_DT_UINT16: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<uint16_t>(), &status);
+                break;
+            }
+            case DHS_DT_UINT32: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<uint32_t>(), &status);
+                break;
+            }
+            case DHS_DT_INT8: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<int8_t>(), &status);
+                break;
+            }
+            case DHS_DT_INT16: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<int16_t>(), &status);
+                break;
+            }
+            case DHS_DT_INT32: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<int32_t>(), &status);
+                break;
+            }
+            case DHS_DT_FLOAT: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<float>(), &status);
+                break;
+            }
+            case DHS_DT_DOUBLE: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<double>(), &status);
+                break;
+            }
+            case DHS_DT_STRING: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<const std::string&>().c_str(), &status);
+                break;
+            }
+            case DHS_DT_BOOLEAN: {
+                dhsAvAdd(avList, iter->getName().c_str(), iter->getType(), 0, NULL,
+                        iter->getValue<bool>(), &status);
+                break;
+            }
+            default: {
+            }
+            }
+        }
+        if(dhsAvListSize(avList, &status) > 0) {
+            pthread_mutex_lock(&lock);
+            DHS_TAG tag = dhsBdPut(connection, id.c_str(), DHS_BD_PT_DS, final?DHS_TRUE:DHS_FALSE, avList, &status);
+            dhsWait(1, &tag, &status);
+            if (dhsStatus(tag, NULL, &status) != DHS_CS_DONE) {
+                status = DHS_E_CTL_CMD;
+            }
+            DHS_STATUS s = DHS_S_SUCCESS;
+            dhsTagFree(tag, &s);
+            pthread_mutex_unlock(&lock);
+        }
+        DHS_STATUS status2 = DHS_S_SUCCESS;
+        dhsAvListFree(avList, &status2);
+    }
+
+    return status;
+}
+
+void DhsAdapter::setTimeout(unsigned int timeout) {
```

Comment:
Timeout implementation for DHS Server replies is still pending. Although, at this point I'm thinking it is not worth it, because:
* The service will be running on the same machine as the DHS server, meaning that IMP communication will be done locally through a memory map, not network.
* Timeouts are not easy to implement, mainly because the DHS functions used don't have timeouts. But, the DHS code catches all kinds of errors and (AFAIK) always produces a reply. That includes detecting a connection problem on the client side.
* I need to implement timeouts on the Seqexec side anyways